oak/server/dev/oak_manager.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2018 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oak_manager.h"
+
+#include "absl/memory/memory.h"
+#include "asylo/grpc/auth/enclave_channel_credentials.h"
+#include "asylo/grpc/auth/enclave_server_credentials.h"
+#include "asylo/grpc/auth/null_credentials_options.h"
+#include "asylo/identity/descriptions.h"
+#include "asylo/identity/init.h"
+#include "asylo/util/logging.h"
+#include "include/grpcpp/grpcpp.h"
+
+namespace oak {
+
+OakManager::OakManager(absl::string_view _path)
+    : Service(), application_id_(0), runtime_(absl::make_unique<OakRuntime>()) {
+  LOG(INFO) << "Creating OakManager";
+}
+
+grpc::Status OakManager::CreateApplication(grpc::ServerContext* context,
+                                           const oak::CreateApplicationRequest* request,
+                                           oak::CreateApplicationResponse* response) {
+  std::string application_id = NewApplicationId();
+  LOG(INFO) << "Creating app with with id: " << application_id;
+
+  InitializeAssertionAuthorities();
+  runtime_->InitializeServer(
+      request->application_configuration(),
+      asylo::EnclaveServerCredentials(asylo::BidirectionalNullCredentialsOptions()));
+
+  response->set_application_id(application_id);
+  response->set_grpc_port(runtime_->GetServerAddress());
+  return grpc::Status::OK;
+}
+
+void OakManager::InitializeAssertionAuthorities() {
+  LOG(INFO) << "Initializing assertion authorities";
+  asylo::EnclaveAssertionAuthorityConfig null_config;
+  asylo::SetNullAssertionDescription(null_config.mutable_description());
```

Comment:
Perhaps add a comment that even though we are not running in an enclave, we can still rely on Asylo assertions and potentially use non-enclave identities (e.g. https://github.com/google/asylo/blob/41d2d39b11b416e54c53ad69d2984a1ce3331b66/asylo/identity/identity.proto#L40)

Reply:
I am liking this less and less :) The whole idea of having a separate target is to not care about Asylo and this is an annoying dependency.

Reply:
Done

Reply:
Look at it from the other point of view: a client configured with asylo credentials will work in both cases, with only changes to the actual assertions to verify.